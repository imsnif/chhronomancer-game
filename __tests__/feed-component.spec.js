import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import serverMock from '../server-mock'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))
jest.useFakeTimers()

beforeEach(() => {
  serverMock()
})

afterEach(() => {
  const clockStore = require('../stores/clock').default
  jest.useRealTimers()
  clockStore.destroy()
  jest.useFakeTimers()
})

test('Feed => renders correctly', () => {
  const Feed = require('../components/feed').default
  const tree = renderer.create(
    <Feed />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
