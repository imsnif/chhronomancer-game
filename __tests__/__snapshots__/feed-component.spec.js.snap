// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#141313",
      "flexDirection": "column",
      "height": "100%",
      "paddingBottom": 5,
      "paddingTop": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 30,
              "marginBottom": 5,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
              }
            }
            style={
              Object {
                "flex": 0,
                "height": 26,
                "margin": 1,
                "resizeMode": "stretch",
                "width": 26,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "green",
                  "height": "100%",
                  "opacity": 0.5,
                  "width": "100%",
                }
              }
            />
          </Image>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "flex-start",
                  "color": "#329932",
                  "flex": 1,
                  "fontFamily": "telegrama_raw",
                  "fontSize": 11,
                  "paddingBottom": 3,
                  "paddingLeft": 3,
                  "paddingRight": 5,
                }
              }
            >
              Aram
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "flex-start",
                  "color": "#329932",
                  "flex": 1,
                  "fontFamily": "telegrama_raw",
                  "fontSize": 9,
                  "marginBottom": 3,
                  "paddingLeft": 3,
                  "paddingRight": 5,
                }
              }
            >
              has travelled here
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#329932",
                  "flex": 1,
                  "fontFamily": "telegrama_raw",
                  "fontSize": 14,
                }
              }
            >
              Timeline 0-1
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#329932",
                  "flex": 1,
                  "fontFamily": "telegrama_raw",
                  "fontSize": 9,
                }
              }
            >
              111
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 30,
              "marginBottom": 5,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
              }
            }
            style={
              Object {
                "flex": 0,
                "height": 26,
                "margin": 1,
                "resizeMode": "stretch",
                "width": 26,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "green",
                  "height": "100%",
                  "opacity": 0.5,
                  "width": "100%",
                }
              }
            />
          </Image>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "flex-start",
                  "color": "#329932",
                  "flex": 1,
                  "fontFamily": "telegrama_raw",
                  "fontSize": 11,
                  "paddingBottom": 3,
                  "paddingLeft": 3,
                  "paddingRight": 5,
                }
              }
            >
              Aram
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "flex-start",
                  "color": "#329932",
                  "flex": 1,
                  "fontFamily": "telegrama_raw",
                  "fontSize": 9,
                  "marginBottom": 3,
                  "paddingLeft": 3,
                  "paddingRight": 5,
                }
              }
            >
              is locking the timeline
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#329932",
                  "flex": 1,
                  "fontFamily": "telegrama_raw",
                  "fontSize": 14,
                }
              }
            >
              Timeline 0-1
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#329932",
                  "flex": 1,
                  "fontFamily": "telegrama_raw",
                  "fontSize": 9,
                }
              }
            >
              111
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
